### PR TITLE
Do not allow negavative time_spent in *Progress services

### LINKED ISSUE #648: Do not allow negative time spent
## Motivations
In the `*Progress` services, when computing time spent, do not allow negative value when validated_at < started_at (that may happen if a chapter is validated through propagation without having been visited (validated through other ancestors)). 

## Subtasks
* Max the time spent with "0" so ensure these case are just considered as "solved but no time actually spent on solving this chapter" which is a correct statement (kind of).

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #648

### CODE REVIEW

File: app/api/groups/get_group_progress.feature [hunk 1]
```diff
@@ -222,6 +222,7 @@ Feature: Display the current progress of a group on a subset of items (groupGrou
       | 0          | 15             | 211     | 2017-04-29 06:38:38 | 0              | null                | 0            | 0           | null                |
       | 0          | 15             | 212     | 2017-03-29 06:38:38 | 0              | null                | 0            | 0           | null                |
       | 0          | 59             | 212     | 2019-01-01 00:00:00 | 10             | null                | 0            | 0           | null                |
+      | 0          | 59             | 214     | 3019-01-01 00:00:00 | 0              | null                | 0            | 0           | null                |
```
Comment (smadbe) [inline app/api/groups/get_group_progress.feature:225]: your test do not really check the output is not <0

File: app/api/groups/get_team_progress.feature [hunk 2]
```diff
@@ -207,6 +207,7 @@ Feature: Display the current progress of teams on a subset of items (groupTeamPr
       | 3          | 14             | 211     | 2017-05-29 06:38:38 | 50             | 2017-05-30 06:38:38 | 10           | 20          | 2017-05-30 06:38:38 | 2018-05-27 06:38:38 |
       | 0          | 15             | 211     | 2017-04-29 06:38:38 | 0              | null                | 0            | 0           | null                | 2018-05-26 06:38:38 |
       | 0          | 15             | 212     | 2017-03-29 06:38:38 | 0              | null                | 0            | 0           | null                | 2018-05-25 06:38:38 |
+      | 0          | 14             | 214     | 3017-03-29 06:38:38 | 0              | null                | 0            | 0           | null                | 2018-05-25 06:38:38 |
```
Comment (smadbe) [inline app/api/groups/get_team_progress.feature:210]: same here

File: app/api/groups/get_user_progress.feature [hunk 3]
```diff
@@ -220,6 +220,7 @@ Feature: Display the current progress of users on a subset of items (groupUserPr
       | 1          | 67             | 212     | 2019-01-01 00:00:00 | 10             | 2017-05-30 06:38:38 | 6            | 7           | null                | 2019-07-01 00:00:00 | # latest_activity_at for 67, 212 comes from this line (the last activity is made by a user)
       | 0          | 67             | 213     | 2018-11-01 00:00:00 | 0              | null                | 0            | 0           | null                | 2018-11-01 00:00:00 | # started_at for 67, 213 comes from this line (the first attempt is started by a user)
       | 0          | 67             | 214     | 2017-05-29 06:38:38 | 15             | 2017-05-29 06:38:48 | 10           | 11          | 2017-05-29 06:38:48 | 2017-05-30 06:38:48 | # min(validated_at) for 67, 214 comes from this line (from a user)
+      | 0          | 67             | 215     | 3018-11-01 00:00:00 | 0              | null                | 0            | 0           | null                | 2018-11-01 00:00:00 | # started_at for 67, 213 comes from this line (the first attempt is started by a user)
```
Comment (smadbe) [inline app/api/groups/get_user_progress.feature:223]: same here

File: app/api/groups/get_group_progress.feature (hunk 1 above)
Comment (zenovich) [reply app/api/groups/get_group_progress.feature:225]: That's not true. I wrote the test first.

**Review by smadbe — APPROVED** [review]
Ah you changed an entry which was already outputted? ok then

**Comment by zenovich** [comment]
> Ah you changed an entry which was already outputted? ok then

exactly

**Pull request merged**